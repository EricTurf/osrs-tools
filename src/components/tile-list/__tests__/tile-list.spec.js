import React, { createContext, useContext as mockUseContext } from 'react';
import { render, fireEvent } from 'react-testing-library';

import TileList from '../tile-list';
const mockDispatch = createContext();

jest.mock('../../../app', () => ({ useDispatch: jest.fn(() => mockUseContext(mockDispatch)) }));

describe('Tile List', () => {
    describe('rendering', () => {
        it('should match the default snapshot', () => {
            const { container } = render(<TileList />);
            expect(container).toMatchSnapshot();
        });
    });

    describe('Actions', () => {
        it('should call the dispatch with the proper page change action from context when a tile is clicked', () => {
            const Context = mockDispatch;
            const dispatch = jest.fn();
            const { getByText } = render(
                <Context.Provider value={dispatch}>
                    <TileList />
                </Context.Provider>
            );

            fireEvent.click(getByText('Herblore Calculator'));
            expect(dispatch).toHaveBeenCalledTimes(1);
            expect(dispatch).toHaveBeenCalledWith({ type: 'PAGE_CHANGE', page: 'herbloreCalc' });

            fireEvent.click(getByText('Ensouled Head Calculator'));
            expect(dispatch).toHaveBeenCalledTimes(2);
            expect(dispatch).toHaveBeenCalledWith({ type: 'PAGE_CHANGE', page: 'ensouledHeadCalc' });
        });
    });
});
