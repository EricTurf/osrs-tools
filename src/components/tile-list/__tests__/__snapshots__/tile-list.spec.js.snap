// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile List rendering should match the default snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa gkTLIw"
  >
    <div
      class="sc-bwzfXH ifanNJ"
    >
      <div
        class="sc-htpNat loqVhN"
      >
        <h2
          class="sc-bZQynM eoBPAX"
        >
          Herblore Calculator
        </h2>
      </div>
      <div
        class="sc-bxivhb huOraT"
      >
        Calculate how much herblore exp you have with your herbs or calculate the amount of money required to gain a certain amount of exp
      </div>
    </div>
    <div
      class="sc-bwzfXH ifanNJ"
    >
      <div
        class="sc-htpNat loqVhN"
      >
        <h2
          class="sc-bZQynM eoBPAX"
        >
          Ensouled Head Calculator
        </h2>
      </div>
      <div
        class="sc-bxivhb huOraT"
      >
        Calculate how much prayer and magic exp you will gain with all your ensouled heads
      </div>
    </div>
  </div>
</div>
`;
